src/palace/manager/api/odl/api.py

```diff
@@ -564,43 +484,66 @@ def _find_matching_delivery_mechanism(
         )
         if fulfillment is None:
             raise FormatNotAvailable()
-        return fulfillment
 
-    def _lcp_fulfill(
+    def _unlimited_access_fulfill(
+        self, loan: Loan, delivery_mechanism: LicensePoolDeliveryMechanism
+    ) -> Fulfillment:
+        licensepool = loan.license_pool
+        self._check_delivery_mechanism_available(
+            delivery_mechanism.delivery_mechanism, licensepool
+        )
+        content_link = delivery_mechanism.resource.representation.public_url
+        content_type = delivery_mechanism.resource.representation.media_type
+        return RedirectFulfillment(content_link, content_type)
+
+    def _license_fulfill(
         self, loan: Loan, delivery_mechanism: LicensePoolDeliveryMechanism
     ) -> Fulfillment:
-        doc = self.get_license_status_document(loan)
-        status = doc.get("status")
+        # We are unable to fulfill a loan that doesn't have its external identifier set,
+        # since we use this to get to the checkout link. It shouldn't be possible to get
+        # into this state.
+        license_status_url = loan.external_identifier
+        assert license_status_url is not None
 
-        if status not in [self.READY_STATUS, self.ACTIVE_STATUS]:
+        doc = self._request_loan_status("GET", license_status_url)
+
+        if not doc.active:
             # This loan isn't available for some reason. It's possible
             # the distributor revoked it or the patron already returned it
             # through the DRM system, and we didn't get a notification
             # from the distributor yet.
             self.update_loan(loan, doc)
             raise CannotFulfill()
 
-        expires = doc.get("potential_rights", {}).get("end")
-        expires = dateutil.parser.parse(expires)
-
-        links = doc.get("links", [])
-
-        content_link, content_type = self._find_content_link_and_type(
-            links, delivery_mechanism.delivery_mechanism.drm_scheme
-        )
+        drm_scheme = delivery_mechanism.delivery_mechanism.drm_scheme
+        fulfill_cls: Callable[[str, str | None], UrlFulfillment]
+        if drm_scheme == DeliveryMechanism.NO_DRM:
+            # If we have no DRM, we can just redirect to the content link and let the patron download the book.
+            fulfill_link = doc.links.get(
+                rel="publication",
+                type=delivery_mechanism.delivery_mechanism.content_type,
+            )
+            fulfill_cls = RedirectFulfillment
```

Comment:
This supports the no-DRM case: 
https://www.notion.so/lyrasis/OPDS2-ODL-Feed-without-LCP-a69bb6024fce4a71b50280d9fa8a4e84